x-pack/legacy/plugins/maps/public/connected_components/map/mb/tooltip_control/tooltip_control.js

```diff
@@ -0,0 +1,296 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import _ from 'lodash';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { I18nProvider } from '@kbn/i18n/react';
+import {
+  FEATURE_ID_PROPERTY_NAME,
+  LON_INDEX,
+} from '../../../../../common/constants';
+import mapboxgl from 'mapbox-gl';
+import { FeatureTooltip } from '../../feature_tooltip';
+
+const TOOLTIP_TYPE = {
+  HOVER: 'HOVER',
+  LOCKED: 'LOCKED'
+};
+
+function justifyAnchorLocation(mbLngLat, targetFeature) {
+  let popupAnchorLocation = [mbLngLat.lng, mbLngLat.lat]; // default popup location to mouse location
+  if (targetFeature.geometry.type === 'Point') {
+    const coordinates = targetFeature.geometry.coordinates.slice();
+
+    // Ensure that if the map is zoomed out such that multiple
+    // copies of the feature are visible, the popup appears
+    // over the copy being pointed to.
+    while (Math.abs(mbLngLat.lng - coordinates[LON_INDEX]) > 180) {
+      coordinates[0] += mbLngLat.lng > coordinates[LON_INDEX] ? 360 : -360;
+    }
+
+    popupAnchorLocation = coordinates;
+  }
+  return popupAnchorLocation;
+}
+
+export class TooltipControl extends React.Component {
+
+  constructor() {
+    super();
+    this._areMapHandlersRegistered = false;
+    this._tooltipContainer = document.createElement('div');
+    this._mbPopup = new mapboxgl.Popup({
+      closeButton: false,
+      closeOnClick: false,
+      maxWidth: '260px', // width of table columns max-widths plus all padding
+    });
+  }
+
+  componentDidUpdate() {
+    if (this.props.mbMap) {
+      this._syncTooltipState();
+
+      if (!this._areMapHandlersRegistered) {
```

Comment:
Couldn't you just put these event handlers in `componentDidMount` and remove the condition?